File: ruby/difference-of-squares/difference_of_squares.rb
class Squares
  VERSION = 2
  def initialize(n)
    @n = n
  end

  def square_of_sum
    return 0 if @n == 0
    sum = (1..@n).reduce(:+)
    sum**2
  end

  def difference
    square_sum = square_of_sum
    sum_square = sum_of_squares

    square_sum - sum_square
  end
  def sum_of_squares
    return 0 if @n == 0
    count = 0
    @n.times do |num|
      count += (num+1)*(num+1)
    end
    return count
  end

end

File: go/difference-of-squares/difference_of_squares.go
package diffsquares

import "math"

const (
	testVersion = 1
)

func SquareOfSums(n int) int {
	sum := 0
	for i := 1; i <= n; i++ {
		sum += i
	}

	return sum * sum
}

func SumOfSquares(n int) int {
	sum := 0
	for i := 1; i <= n; i++ {
		sum += i * i
	}
	return sum
}

func Difference(n int) int {
	return int(math.Abs(float64(SumOfSquares(n)) - float64(SquareOfSums(n))))
}
